doc/apiv3.apib

```diff
@@ -157,12 +154,330 @@ he is allowed to see his own time entries in projects where the costs module is
+
+    **Required permission:** view cost entries **or** view own cost entries (on cost entry's project)
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:MissingPermission",
+                "message": "You are not allowed to see this cost entry."
+            }
+
+## Cost Entries by work package [/api/v3/work_packages/{id}/cost_entries]
+
++ Model
+    + Body
+
+            {
+                "_links":
+                {
+                    "self":
+                    {
+                        "href": "/api/v3/work_packages/1/cost_entries"
+                    }
+                },
+                "total": 1,
+                "count": 1,
+                "_type": "Collection",
+                "_embedded":
+                {
+                    "elements": [
+                        {
+                            "_type": "CostEntry",
+                            "_links": {
+                                "self": {
+                                    "href": "/api/v3/cost_entries/1"
+                                },
+                                "project": {
+                                    "href": "/api/v3/projects/1"
+                                },
+                                "costType": {
+                                    "href": "/api/v3/cost_types/1"
+                                },
+                                "user": {
+                                    "href": "/api/v3/users/1"
+                                },
+                                "workPackage": {
+                                    "href": "/api/v3/work_packages/1"
+                                }
+                            },
+                            "id": 1,
+                            "spentUnits": 3.14,
+                            "spentOn": "2015-03-31",
+                            "createdAt": "2015-03-31T08:51:20Z",
+                            "updatedAt": "2015-03-31T08:51:20Z"
+                        }
+                    ]
+                }
+            }
+
+## list Cost Entries of a work package [GET]
+
++ Parameters
+    + id (required, integer, `1`) ... work package id
+
++ Response 200 (application/hal+json)
+
+    [Cost Entries by work package][]
+
++ Response 403 (application/hal+json)
+
+    Returned if the client does not have sufficient permissions.
+
+    **Required permission:** view cost entries **or** view own cost entries (on work package's project)
+
+    *Note that you will only receive this error, if you are at least allowed to see the corresponding work package.*
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:MissingPermission",
+                "message": "You are not allowed to see the cost entries of this work package."
+            }
+
++ Response 404 (application/hal+json)
```

Comment:
This response cannot be provoked as the necessary set of circumstances result in a 403.